### PR TITLE
Strapi Integration

### PR DESCRIPTION
**Pull request by partha6686**

### CODE REVIEW

File: components/blog/BlogCard.vue [hunk 1]
```diff
@@ -7,7 +7,7 @@
       :title="article.coverImgAlt"
     />
     <div class="d-flex flex-column align-items-start">
-      <span class="blog__date mt-3">{{ formatDate(article.createdAt) }}</span>
+      <span class="blog__date mt-3">{{ formatDate(article.publishedAt) }}</span>
```
Comment (ankitsinghaniyaz) [inline components/blog/BlogCard.vue:10]: Published date will be the same that it used to be right? It will not change the date after migration I hope?

Comment (partha6686) [reply components/blog/BlogCard.vue:10]: Yes Yes it is same as before..

File: pages/blog/_slug.vue [hunk 2]
```diff
@@ -150,16 +154,34 @@ export default {
   content: {
     liveEdit: false,
   },
-  async asyncData({ $content, params }) {
-    const article = await $content('blog', params.slug).fetch()
+  async asyncData({ params }) {
+    const response = await axios.get(
```
Comment (ankitsinghaniyaz) [inline pages/blog/_slug.vue:158]: I think there's a way in axios to pass params. you don't have to do ?something=dsf manually

File: pages/blog/_slug.vue [hunk 3]
```diff
@@ -150,16 +154,34 @@ export default {
...
+      `${process.env.strapiUrl}/api/blogs?filters[slug][$eq]=${params.slug}&populate=*`
+    )
+    const readingStats = readingTime(response.data.data[0].attributes.body)
```
Comment (ankitsinghaniyaz) [inline pages/blog/_slug.vue:161]: should not be doing `response.data.data[0]` at multiple places. Rather set a variable blog = response.data.data[0] and then use that

File: pages/blog/index.vue [hunk 4]
```diff
@@ -23,29 +23,33 @@ import BlogFeatured from '../../components/blog/BlogFeatured.vue'
...
-        published: true,
-        featured: false,
-      })
-      .sortBy('createdAt', 'desc')
-      .fetch()
+  async asyncData() {
+    const {
+      data: { data },
+    } = await axios.get(
+      `${process.env.strapiUrl}/api/blogs?sort=publishedAt:desc&populate=*`
```
Comment (ankitsinghaniyaz) [inline pages/blog/index.vue:39]: same

File: pages/blog/_slug.vue (hunk 2 above)
Comment (partha6686) [reply pages/blog/_slug.vue:158]: Okay, I will change this in all places..

File: pages/use-case/human-resource.vue [hunk 5]
```diff
@@ -53,18 +57,22 @@ export default {
...
+        ...item.attributes,
+        coverImg: item.attributes.coverImg.data.attributes.url,
+        readingStats: readingTime(item.attributes.body),
+      }
+    })
 
-    const randIndex = Math.floor(Math.random() * (articles.length - 3))
-    articles = articles.slice(randIndex, randIndex + 3)
+    const randIndex = Math.floor(Math.random() * (articles.length - 3));
+    articles = articles?.slice(randIndex, randIndex + 3);
```
Comment (ankitsinghaniyaz) [inline pages/use-case/human-resource.vue:75]: I see this is being used at multiple places. It's good idea to create a component called as `<related-blogs :slug="slug" count="3" />` which takes care of this thing in one component and renders it.

**Review by ankitsinghaniyaz — APPROVED** [review]

File: nuxt.config.js [hunk 6]
```diff
@@ -109,15 +111,6 @@ export default {
...
 
-  // Hooks configuration - https://content.nuxtjs.org/advanced/
-  hooks: {
-    'content:file:beforeInsert': (document) => {
-      if (document.extension === '.md') {
-        const stats = require('reading-time')(document.text)
-        document.readingStats = stats
-      }
-    },
-  },
```
Comment (piyushsinghania) [inline nuxt.config.js:120]: How are we handling reading time now?

File: pages/blog/_slug.vue [hunk 7]
```diff
@@ -2,20 +2,21 @@
   <div>
     <div class="container position-relative">
       <article
+        v-if="blogData"
         class="container mw-920 mt-8rem"
-        :class="{ 'mb-3rem': !(article.cta && article.cta.hidden) }"
+        :class="{ 'mb-3rem': !(blogData?.cta && blogData.cta.hidden) }"
```
Comment (piyushsinghania) [inline pages/blog/_slug.vue:7]: After having `v-if` do we still need `?`for `blogData?.cta` etc?

File: pages/blog/_slug.vue [hunk 8]
```diff
@@ -150,16 +154,42 @@ export default {
...
   },
-  async asyncData({ $content, params }) {
-    const article = await $content('blog', params.slug).fetch()
+  async asyncData({ params }) {
+    const response = await axios.get(`${process.env.strapiUrl}/api/blogs`, {
+      params: {
+        'filters[slug][$eq]': params.slug,
+        populate: '*',
+      },
+    })
```
Comment (piyushsinghania) [inline pages/blog/_slug.vue:163]: Since here, we are making a network call, a `try-catch`  block might help.

File: pages/blog/_slug.vue [hunk 9]
```diff
@@ -223,18 +253,19 @@ export default {
   },
   computed: {
     meta() {
+      // console.log("this.blogData?.coverImg", this.blogData?.coverImg);
```
Comment (piyushsinghania) [inline pages/blog/_slug.vue:256]: Maybe not needed any more?

File: nuxt.config.js (hunk 6 above)
Comment (partha6686) [reply nuxt.config.js:120]: When fetching the data at that time we are calculating the reading time using the `reading-time` package.

File: pages/blog/index.vue [hunk 10]
```diff
@@ -23,29 +23,36 @@ import BlogFeatured from '../../components/blog/BlogFeatured.vue'
...
-      })
-      .sortBy('createdAt', 'desc')
-      .fetch()
+  async asyncData() {
+    const {
+      data: { data },
+    } = await axios.get(`${process.env.strapiUrl}/api/blogs`, {
+      params: {
+        sort: 'publishedAt:desc',
+        populate: '*',
```
Comment (piyushsinghania) [inline pages/blog/index.vue:41]: What's populate `*`?

File: pages/use-case/human-resource.vue [hunk 11]
```diff
@@ -53,18 +56,32 @@ export default {
...
+        },
+      }
+    )
+
+    let articles = data.map((item) => {
+      return {
+        id: item.id,
+        ...item.attributes,
+        coverImg: item.attributes.coverImg.data.attributes.url,
+        readingStats: readingTime(item.attributes.body),
```
Comment (piyushsinghania) [inline pages/use-case/human-resource.vue:79]: We might need to document the `attributes` somewhere?

File: pages/blog/index.vue (hunk 10 above)
Comment (partha6686) [reply pages/blog/index.vue:41]: We need to pass this to populate the deeply nested fields like cover images and meta Images..
These fields are not populated by default.

File: utils/getRoutes.js [hunk 12]
```diff
@@ -1,7 +1,13 @@
 export default async () => {
-  const { $content } = require("@nuxt/content");
-  let files = await $content({ deep: true }).only(["path", "published"]).fetch();
-  files = files.filter(file => file.published)
+  const axios = require('axios')
+  
+  const {
+    data: { data },
+  } = await axios.get(`http://localhost:1337/api/blogs?populate=*`);
```
Comment (piyushsinghania) [inline utils/getRoutes.js:6]: `Todo:` ~localhost~

File: pages/use-case/human-resource.vue (hunk 11 above)
Comment (partha6686) [reply pages/use-case/human-resource.vue:79]: Sorry, I didn't get it..

File: utils/getRoutes.js (hunk 12 above)
Comment (ankitsinghaniyaz) [reply utils/getRoutes.js:6]: I think we missed this one.

File: pages/blog/_slug.vue (hunk 7 above)
Comment (ankitsinghaniyaz) [reply pages/blog/_slug.vue:7]: I think it's fine

File: utils/getRoutes.js (hunk 12 above)
Comment (partha6686) [reply utils/getRoutes.js:6]: yes

File: pages/blog/_slug.vue (hunk 8 above)
Comment (ankitsinghaniyaz) [reply pages/blog/_slug.vue:163]: good point, though what are you going to do after catching ?

File: pages/blog/_slug.vue (hunk 7 above)
Comment (piyushsinghania) [reply pages/blog/_slug.vue:7]: I see! Also, I think, the `v-if`  will eliminate the need of using `?.`

Comment (piyushsinghania) [reply pages/blog/_slug.vue:7]: Anyway, it's okay for now!

File: pages/blog/_slug.vue (hunk 8 above)
Comment (piyushsinghania) [reply pages/blog/_slug.vue:163]: Maybe we can redirect to blog index page.

File: pages/blog/index.vue (hunk 10 above)
Comment (piyushsinghania) [reply pages/blog/index.vue:41]: I see, thanks for explaining.

File: pages/use-case/human-resource.vue (hunk 11 above)
Comment (piyushsinghania) [reply pages/use-case/human-resource.vue:79]: Yes, we can be okay with it.

**Pull request merged**